source/MaterialXGenShader/Nodes/SourceCodeNode.cpp

```diff
@@ -71,11 +71,18 @@ void SourceCodeNode::emitFunctionDefinition(const ShaderNode&, GenContext& conte
 {
     BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
         // Emit function definition for non-inlined functions
-        if (!_inlined && !_functionSource.empty())
+        if (!_functionSource.empty())
         {
-            const ShaderGenerator& shadergen = context.getShaderGenerator();
-            shadergen.emitBlock(_functionSource, _sourceFilename, context, stage);
-            shadergen.emitLineBreak(stage);
+            if (!_sourceFilename.isEmpty())
+            {
+                stage.addSourceDependency(_sourceFilename);
```

Comment:
Changed to a FilePath

Reply:
Note: I kept _sourceDependencies a StringSet because that's what _includes does, and I don't see an example of storing FilePath in std::set or std::map so wasn't sure it was set up to allow that.